Dockerfile

```diff
@@ -1,58 +1,74 @@
-FROM rust:slim
+# To produce a smaller image this Dockerfile contains two separate stages: in
+# the first one all the build dependencies are installed and docs.rs is built,
+# while in the second one just the runtime dependencies are installed, with the
+# binary built in the previous stage copied there.
+#
+# As of 2019-10-29 this reduces the image from 2.8GB to 500 MB.
 
-### STEP 1: Install dependencies ###
-# Install packaged dependencies
-RUN apt-get update && apt-get install -y --no-install-recommends \
-  build-essential git curl cmake gcc g++ pkg-config libmagic-dev \
-  libssl-dev zlib1g-dev sudo docker.io
+#################
+#  Build stage  #
+#################
 
-### STEP 2: Create user ###
-ENV HOME=/home/cratesfyi
-RUN adduser --home $HOME --disabled-login --disabled-password --gecos "" cratesfyi
+FROM ubuntu:bionic AS build
 
-### STEP 3: Setup build environment as new user ###
-ENV CRATESFYI_PREFIX=/home/cratesfyi/prefix
-RUN mkdir $CRATESFYI_PREFIX && chown cratesfyi:cratesfyi "$CRATESFYI_PREFIX"
+# Install packaged dependencies
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+  build-essential git curl cmake gcc g++ pkg-config libmagic-dev \
+  libssl-dev zlib1g-dev sudo ca-certificates docker.io
```

Comment:
`sudo` and `docker.io` are no longer necessary. Most of the other commands are also not needed I think.